File: sudoku-ums/ums-boot/src/main/java/com/zumin/sudoku/ums/exception/UserException.java
package com.zumin.sudoku.ums.exception;

import com.zumin.sudoku.common.core.exception.BaseException;
import com.zumin.sudoku.ums.enums.UmsStatusCode;
import lombok.Getter;

/**
 * 用户异常类
 */
@Getter
public class UserException extends BaseException {

  private static final long serialVersionUID = -799932046646436382L;

  /**
   * 用户异常类的无参构造方法
   */
  public UserException() {
    super(UmsStatusCode.USER_ERROR);
  }

  /**
   * 用户异常类的构造方法
   *
   * @param statusCode 状态编码
   */
  public UserException(UmsStatusCode statusCode) {
    super(statusCode);
  }

}

File: sudoku-ums/ums-boot/src/main/java/com/zumin/sudoku/ums/exception/UserException.kt
package com.zumin.sudoku.ums.exception

import com.zumin.sudoku.common.core.exception.BaseException
import com.zumin.sudoku.ums.UmsStatusCode

/**
 * 用户异常类
 */
class UserException : BaseException {
  /**
   * 用户异常类的无参构造方法
   */
  constructor() : super(UmsStatusCode.USER_ERROR)

  /**
   * 用户异常类的构造方法
   *
   * @param statusCode 状态编码
   */
  constructor(statusCode: UmsStatusCode) : super(statusCode)

  companion object {
    private const val serialVersionUID = -799932046646436382L
  }
}